### PR TITLE
Test purge ioclass

### PR DESCRIPTION
**Pull request by mmichal10**

### CODE REVIEW

File: test/functional/tests/io_class/test_io_class_purge.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_ioclass_usage_sum():
+    """
+        title: Test for ioclass stats after purge
+        description: |
+          Create ioclasses for 3 different directories. Run IO against each
+          directory, check usage stats correctness before and after purge
+        pass_criteria:
+          - Usage stats are consistent on each test step
+          - Usage stats doesn't exceed cache size
```
Comment (Deixx) [inline test/functional/tests/io_class/test_io_class_purge.py:37]: don't

File: test/functional/tests/io_class/test_io_class_purge.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        TestRun.LOGGER.error(
+            "Sum of ioclasses usage stats doesn't match cache usage stats!"
+            f" Cache stats:\n{cache_usage_stats} ioclasses sum:\n{usage_stats_sum}"
+            f" Stats of particualr ioclasses:\n"
+            f"{[get_io_class_usage(cache, i) for i in ioclasses_ids]}"
+        )
+
+    if cache_usage_stats.occupancy + cache_usage_stats.free > cache_size:
+        TestRun.LOGGER.error(
+            "Sum of occupancy and free cache lines exceed cache size!"
```
Comment (Deixx) [inline test/functional/tests/io_class/test_io_class_purge.py:122]: exceeds

File: test/functional/tests/io_class/test_io_class_purge.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    with TestRun.step("Trigger IO to each partition for the second time and verify stats"):
+        for io_class in io_classes:
+            run_io_dir(io_class.dir_path, int((io_class.io_size) / Unit.Blocks4096))
+
+        verify_ioclass_usage_stats(cache, [i.id for i in io_classes])
+
+
+def get_io_class_usage(cache, io_class_id):
+    return cache.get_io_class_statistics(io_class_id=io_class_id).usage_stats
```
Comment (Ostrokrzew) [inline test/functional/tests/io_class/test_io_class_purge.py:100]: Isn't it overkill? You just shortened the function call.

File: test/functional/tests/io_class/test_io_class_purge.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    for i in ioclasses_ids:
+        usage_stats_sum += get_io_class_usage(cache, i)
+
+    cache_usage_stats = cache.get_statistics().usage_stats
+
+    if usage_stats_sum != cache_usage_stats:
+        TestRun.LOGGER.error(
+            "Sum of ioclasses usage stats doesn't match cache usage stats!"
+            f" Cache stats:\n{cache_usage_stats} ioclasses sum:\n{usage_stats_sum}"
+            f" Stats of particualr ioclasses:\n"
```
Comment (Ostrokrzew) [inline test/functional/tests/io_class/test_io_class_purge.py:116]: typo in 'particular'

File: test/functional/tests/io_class/test_io_class_purge.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+          - Usage stats doesn't exceed cache size
+    """
+    with TestRun.step("Prepare disks"):
+        cache, core = prepare()
+        cache_size = cache.get_statistics().config_stats.cache_size
+
+    with TestRun.step("Disable udev"):
+        Udev.disable()
+
+    with TestRun.step(f"Preparing filesystem and mounting {core.system_path} at {mountpoint}"):
```
Comment (katlapinka) [inline test/functional/tests/io_class/test_io_class_purge.py:38]: Please use the same verbs form in each test step description, e.g. prepare, add, etc.

File: test/functional/tests/io_class/test_io_class_purge.py (hunk 3 above)
Comment (mmichal10) [reply test/functional/tests/io_class/test_io_class_purge.py:92]: I wanted to make it shorter than whole line width.

**Comment by mmichal10** [comment]
retest this please

**Comment by mmichal10** [comment]
retest this please

**Pull request merged**